packages/nextjs-mf/src/plugins/container/CommonJsChunkLoadingPlugin.ts

```diff
@@ -0,0 +1,126 @@
+import type { Chunk, Compiler } from 'webpack';
+//@ts-ignore
+import type { ModuleFederationPluginOptions } from '../types';
+//@ts-ignore
+import RuntimeGlobals from 'webpack/lib/RuntimeGlobals';
+//@ts-ignore
+import StartupChunkDependenciesPlugin from 'webpack/lib/runtime/StartupChunkDependenciesPlugin';
+
+import ChunkLoadingRuntimeModule from './LoadFileChunkLoadingRuntimeModule';
+
+interface CommonJsChunkLoadingOptions extends ModuleFederationPluginOptions {
+  baseURI: Compiler['options']['output']['publicPath'];
+  promiseBaseURI?: string;
+  remotes: Record<string, string>;
+  name?: string;
+  asyncChunkLoading: boolean;
+  verbose?: boolean;
+}
+
+class CommonJsChunkLoadingPlugin {
+  private options: CommonJsChunkLoadingOptions;
+  private _asyncChunkLoading: boolean;
+
+  constructor(options: CommonJsChunkLoadingOptions) {
+    this.options = options || ({} as CommonJsChunkLoadingOptions);
+    this._asyncChunkLoading = this.options.asyncChunkLoading;
+  }
+
+  apply(compiler: Compiler) {
+    const chunkLoadingValue = this._asyncChunkLoading
+      ? 'async-node'
+      : 'require';
+
+    new StartupChunkDependenciesPlugin({
+      chunkLoading: chunkLoadingValue,
+      asyncChunkLoading: this._asyncChunkLoading,
+    }).apply(compiler);
+
+    const { RuntimeGlobals, runtime } = compiler.webpack;
+
+    compiler.hooks.thisCompilation.tap(
+      'CommonJsChunkLoadingPlugin',
+      (compilation) => {
+        // Always enabled
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const isEnabledForChunk = (_: Chunk) => true;
+        const onceForChunkSet = new WeakSet();
+
+        const handler = (chunk: Chunk, set: Set<string>) => {
+          if (onceForChunkSet.has(chunk)) return;
+
+          onceForChunkSet.add(chunk);
+
+          if (!isEnabledForChunk(chunk)) return;
```

Comment:
`isEnabledForChunk` is always returning `true`.
Can this check be removed together with that function?

Reply:
A lot of this can be deleted. I copied a other runtime module and reverse engineered it. Tons of left over stuff